### PR TITLE
Run UI tests in CI for some other crates

### PR DESCRIPTION
**Pull request by gui1117**
The test name is `test-frame-ui` I don't know if I can also change it to `test-ui` without breaking other stuff. So I kept the name unchanged.

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

**Comment by gui1117** [comment]
xcm-procedural ui test are failing to compile because of this: https://github.com/dtolnay/trybuild/pull/282

**Comment by gui1117** [comment]
> @gui1117 can you please change version of `trybuild` in the workspace `Cargo.toml` as well?

yes done in https://github.com/paritytech/polkadot-sdk/pull/5167/commits/47d75df26b238f869446e3573dd793b34879f6bc

File: .gitlab/pipeline/test.yml [hunk 1]
```diff
@@ -398,6 +398,10 @@ test-frame-ui:
   script:
     - time cargo test --locked -q --profile testnet -p frame-support-test --features=frame-feature-testing,no-metadata-docs,try-runtime,experimental
     - time cargo test --locked -q --profile testnet -p frame-support-test --features=frame-feature-testing,frame-feature-testing-2,no-metadata-docs,try-runtime,experimental
+    - time cargo test --locked -q --profile testnet -p xcm-procedural
+    - time cargo test --locked -q --profile testnet -p frame-election-provider-solution-type
+    - time cargo test --locked -q --profile testnet -p sp-api-test
+    - time cargo test --locked -q --profile testnet -p sp-runtime-interface
```
Comment (ggwpez) [inline .gitlab/pipeline/test.yml:404]: I think you can pass the `-p` flag multiple times.

Comment (bkchr) [reply .gitlab/pipeline/test.yml:404]: Yeah, but this way it is better readable, IMO.

**Review by ggwpez — APPROVED** [review]

**Comment by bkchr** [comment]
@gui1117 could you update the UI tests? :D

**Comment by ggwpez** [comment]
bot update-ui

**Comment by mordamax** [comment]
> bot update-ui
> 
> (Lets try, if it does not work, then we have to fix the bot)

it did start ok, https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/6847716

**Comment by gui1117** [comment]
The script fails because a test in frame-support-test compiles while it should fail to compile.
But the CI was working and executed this test.

I tried to set my environment same as the CI but I always failed to reproduce the CI behavior.
So I'm clueless how did the CI make it work.

Writing in `pallet_ui.rs` this line doesn't work
```
	// Deny all warnings since we emit warnings as part of a Pallet's UI.
	std::env::set_var("RUSTFLAGS", "--deny warnings");
```

maybe we should just write `#[deny(warnings)]` inside the test itself.

**Comment by gui1117** [comment]
new version of trybuild seems to break the CI.

It seems that "--deny warnings" is not propagated anymore.

I guess because of this: https://github.com/dtolnay/trybuild/pull/278/files#r1696208255

I will try to fix trybuild, but we can also modify all pass test and the relevant fail tests to have `#[deny(warnings)]`

**Comment by ggwpez** [comment]
> I will try to fix trybuild, but we can also modify all pass test and the relevant fail tests to have `#[deny(warnings)]`

Hm, so the `SKIP_WASM_BUILD` will also not be supported? Then maybe roll back the update of trybuild until it is fixed upstream.

**Comment by gui1117** [comment]
I reverted to old trybuild version and explicity set `std` as a feature so we don't need to rely on the fix: https://github.com/paritytech/polkadot-sdk/pull/5167/commits/092ffbc1d99e2f404360f49a27b3bc4869650f29

**Comment by gui1117** [comment]
> Hm, so the `SKIP_WASM_BUILD` will also not be supported? Then maybe roll back the update of trybuild until it is fixed upstream.

`SKIP_WASM_BUILD` should work, only `RUSTFLAGS` are removed from the environment.

**Review by ggwpez — APPROVED** [review]

**Pull request merged**